### PR TITLE
Add Strassen Matrix Multiplication algorithm

### PR DESCRIPTION
**Pull request by hoplite2000**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

#### References
Fixes issue #2488 

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by hoplite2000** [comment]
Hello @siriak! Your review is much appreciated.

File: DivideAndConquer/StrassenMatrixMultiplication.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            int[][] B21 = new int[n / 2][n / 2];
+            int[][] B22 = new int[n / 2][n / 2];
+ 
+            // Dividing matrix A into 4 halves
+            split(A, A11, 0, 0);
+            split(A, A12, 0, n / 2);
+            split(A, A21, n / 2, 0);
+            split(A, A22, n / 2, n / 2);
+ 
+            // Dividing matrix B into 4 halves
```
Comment (siriak) [inline DivideAndConquer/StrassenMatrixMultiplication.java:36]:
```suggestion
            // Dividing matrix B into 4 parts
```
4 halves? I thought there are only 2 of them...

File: DivideAndConquer/StrassenMatrixMultiplication.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+            int[][] A11 = new int[n / 2][n / 2];
+            int[][] A12 = new int[n / 2][n / 2];
+            int[][] A21 = new int[n / 2][n / 2];
+            int[][] A22 = new int[n / 2][n / 2];
+            int[][] B11 = new int[n / 2][n / 2];
+            int[][] B12 = new int[n / 2][n / 2];
+            int[][] B21 = new int[n / 2][n / 2];
+            int[][] B22 = new int[n / 2][n / 2];
+ 
+            // Dividing matrix A into 4 halves
```
Comment (siriak) [inline DivideAndConquer/StrassenMatrixMultiplication.java:30]:
```suggestion
            // Dividing matrix A into 4 parts
```

**Review by siriak — CHANGES_REQUESTED** [review]
LGTM, only one comment :)

**Comment by siriak** [comment]
Build is failing
./DivideAndConquer/StrassenMatrixMultiplication.java:4: error: class Strassen is public, should be declared in a file named Strassen.java
public class Strassen {

**Review by siriak — APPROVED** [review]

**Pull request merged**